src/Raven.Server/Dashboard/DatabasesInfoNotificationSender.cs

```diff
@@ -148,12 +146,48 @@ public static IEnumerable<AbstractDashboardNotification> FetchDatabasesInfo(Serv
                         };
                         trafficWatch.Items.Add(trafficWatchItem);
 
-                        foreach (var mountPointUsage in database.GetMountPointsUsage())
+                        var currentEnvironmentsHash = database.GetEnvironmentsHash();
+                        if (CachedDatabaseInfo.TryGetValue(databaseName, out var cached) && cached.Hash == currentEnvironmentsHash)
                         {
-                            if (cts.IsCancellationRequested)
-                                yield break;
+                            databasesInfo.Items.Add(cached.Item);
 
-                            UpdateMountPoint(mountPointUsage, databaseName, drivesUsage);
+                            if (cached.NextDiskSpaceCheck < DateTime.UtcNow)
+                            {
+                                CachedDatabaseInfo[databaseName].MountPoints = new List<Client.ServerWide.Operations.MountPointUsage>();
```

Comment:
Actually, you need to use cached.MountPoints here and pass it to the DiskUsageCheck to avoid concurrency issues

Reply:
Passing the DatabaseInfoCache instance should suffice or your case